src/store/selection/actions.ts

```diff
@@ -0,0 +1,26 @@
+import { createAction, PayloadAction } from '@reduxjs/toolkit';
+import { DOMRectMini } from '../../@types';
+import { SelectionItem } from './types';
+
+export const domRectToMini = ({ height, width, x, y }: DOMRect): DOMRectMini => ({
+    height,
+    width,
+    x,
+    y,
+});
+
+export type CreateArg = {
+    location: number;
+    range: Range;
+};
+
+export const setSelectionAction = createAction<SelectionItem | null>('SET_SELECTION');
+export const createSelectionAction = (arg: CreateArg): PayloadAction<SelectionItem | null> => {
```

Comment:
This can be simplified by using the `prepare` [callback parameter](https://redux-toolkit.js.org/api/createAction#using-prepare-callbacks-to-customize-action-contents) for `setSelectionAction`.